base/src/main/java/com/smartdevicelink/managers/screen/choiceset/BaseChoiceSetManager.java

```diff
@@ -461,6 +465,7 @@ public void setKeyboardConfiguration(@Nullable KeyboardProperties keyboardConfig
             properties.setKeypressMode((keyboardConfiguration.getKeypressMode() == null ? KeypressMode.RESEND_CURRENT_ENTRY : keyboardConfiguration.getKeypressMode()));
             properties.setLimitedCharacterList(keyboardConfiguration.getLimitedCharacterList());
             properties.setAutoCompleteText(keyboardConfiguration.getAutoCompleteText());
+            properties.setMaskInputCharacters(keyboardConfiguration.getMaskInputCharacters());
```

Comment:
After checking the proposal, I think that this is missing Android and iOS changes # 4. Here's how I think that would work:

1. The manager needs to handle what happens if the keyboard properties that the developer is trying to use are not supported.
2. If the `KeyboardProperties` that the developer is attempting to use an unsupported `KeyboardLayout`, the operation will need to fail with an error log.
3. If the `KeyboardProperties` that the developer is attempting to use uses too many `customKeys` for the current layout, we will shorten the amount they send to be the maximum amount and print a warning log.
4. If the `KeyboardProperties` that the developer is attempting to use uses a `customKeys` array that is empty, we will remove the array to nil.
5. If the `KeyboardProperties` that the developer is attempting to use uses `maskInputCharacters` when that is not supported on the current layout, we will remove the property to nil.
6. If the `KeyboardCapabilities` are not available, use the existing behavior of sending whatever the developer set.
7. Note that `KeyboardProperties` can be passed by developer in both `setKeyboardConfiguration()` && `presentKeyboard()` so we need to make sure that the properties are handled properly in both places.